### DISCUSSION TITLE
Errors with Install openmm_deepmd_plugin

### DISCUSSION THREAD

**Discussion by mdionic** [deepmodeling/deepmd-kit#2595] [Q&A]
Dear all, 
     Hi, I want to install the dp plugin for OpenMM for its better supports GPUs. And I follow the instructions according to the link (https://github.com/JingHuangLab/openmm_deepmd_plugin). Everything went well until the command  "**make test**“ in the 6th step. The erros are:
"
    Start 1: TestSerializeDeepmdForce
1/5 Test #1: TestSerializeDeepmdForce .........***Failed    0.51 sec
    Start 2: TestDeepmdPlugin4Reference
2/5 Test #2: TestDeepmdPlugin4Reference .......***Failed    0.44 sec
    Start 3: TestDeepmdPlugin4CUDASingle
3/5 Test #3: TestDeepmdPlugin4CUDASingle ......***Failed    0.50 sec
    Start 4: TestDeepmdPlugin4CUDAMixed
4/5 Test #4: TestDeepmdPlugin4CUDAMixed .......***Failed    0.50 sec
    Start 5: TestDeepmdPlugin4CUDADouble
5/5 Test #5: TestDeepmdPlugin4CUDADouble ......***Failed    0.50 sec
0% tests passed, 5 tests failed out of 5
Total Test time (real) =   2.46 sec
The following tests FAILED:
	  1 - TestSerializeDeepmdForce (Failed)
	  2 - TestDeepmdPlugin4Reference (Failed)
	  3 - TestDeepmdPlugin4CUDASingle (Failed)
	  4 - TestDeepmdPlugin4CUDAMixed (Failed)
	  5 - TestDeepmdPlugin4CUDADouble (Failed)
Errors while running CTest
make: *** [test] Error 8" 
And output of the command **./TestDeepmdPlugin4CUDA**   is:
"I tensorflow/core/util/port.cc:110] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
DeePMD-kit: Successfully load libcudart.so
OpenMMException: Graph file not found: ../tests/frozen_model/water.pb"

The **OpenMM version is 8.0**, I also tested python with many versions (including **3.11,3.10,3.8**), but it did not work.
Could anyone give me some suggestions?

**Comment by njzjz** [comment]
> DeePMD-kit: Successfully load libcudart.so OpenMMException: Graph file not found: ../tests/frozen_model/water.pb"

It seems to me that the model file does not exist or is not put into the correct directory. CC @dingye18

**Comment by dingye18** [comment]
Sorry, it's my fault. I did delete the `tests` directory several days ago. Currently, `water.pb` file is placed in `python/OpenMMDeepmdPlugin/data/water.pb`. I have update the path to water.pb in these test files in the latest [PR](https://github.com/JingHuangLab/openmm_deepmd_plugin/pull/14). Can you try again with the latest code in OpenMM Deepmd Plugin?

**Reply by mdionic** [reply]
Thank you for the reply. But some new errors occurs. 
"    Start 1: TestSerializeDeepmdForce
1/5 Test #1: TestSerializeDeepmdForce .........***Failed    0.49 sec
    Start 2: TestDeepmdPlugin4Reference
2/5 Test #2: TestDeepmdPlugin4Reference .......***Failed    0.44 sec
    Start 3: TestDeepmdPlugin4CUDASingle
3/5 Test #3: TestDeepmdPlugin4CUDASingle ......Child aborted***Exception:   0.97 sec
    Start 4: TestDeepmdPlugin4CUDAMixed
4/5 Test #4: TestDeepmdPlugin4CUDAMixed .......Child aborted***Exception:   0.61 sec
    Start 5: TestDeepmdPlugin4CUDADouble
5/5 Test #5: TestDeepmdPlugin4CUDADouble ......Child aborted***Exception:   0.53 sec
0% tests passed, 5 tests failed out of 5
Total Test time (real) =   3.08 sec
The following tests FAILED:
	  1 - TestSerializeDeepmdForce (Failed)
	  2 - TestDeepmdPlugin4Reference (Failed)
	  3 - TestDeepmdPlugin4CUDASingle (Child aborted)
	  4 - TestDeepmdPlugin4CUDAMixed (Child aborted)
	  5 - TestDeepmdPlugin4CUDADouble (Child aborted)
Errors while running CTest
make: *** [test] Error 8"
and the output for the command **./TestDeepmdPlugin4CUDA**   is,
"DeePMD-kit: Successfully load libcudart.so
DATA_LOSS: Can't parse ../python/OpenMMDeepmdPlugin/data/water.pb as binary proto
terminate called after throwing an instance of 'deepmd::hpp::deepmd_exception'
  what():  DeePMD-kit C API Error: DeePMD-kit Error: TensorFlow Error: DATA_LOSS: Can't parse ../python/OpenMMDeepmdPlugin/data/water.pb as binary proto
Aborted (core dumped)"
What are the pre-requirements before the installation? I also want to know if the current version for the python(3.8), OpenMM(8.0) is OK? Do I need to install the TensorFlow before the installation? If so, what is the suitable version ? 

Thank you in advance!

**Reply by dingye18** [reply]
The current version is ok (both python and OpenMM, I did test on them before). TensorFlow is not needed when you using this plugin for simulation. Library of TensorFlow is already installed in libdeepmd_c.

**Reply by dingye18** [reply]
Since the water.pb file is stored in git repo by `git lfs`. The file `water.pb` need to pull by `git lfs pull` in this repo. To use the `git lfs` command, you need to install `git-lfs` through this [link](https://github.com/git-lfs/git-lfs/blob/main/INSTALLING.md) first. After installation , try to run `git lfs pull` in the plugin repo directory. And run `make test` again.

Sorry that I didn’t realize this problem before. I will update the readme later.
